sample-apps/README.md

```diff
@@ -38,3 +38,245 @@ List of sample apps across all repositories in [aws-observability](https://githu
+“metricType”: (string)“request”
+}
+```
+
+**Random:**
+
+Counter
+
+* Name: timeAlive
+* Unit: “ms”
+* Description: “Total amount of time that the application has been alive”
+
+Asynchronous UpDown Counter
+
+* Name: totalHeapSize
+* Unit: “By”
+* Description: “The current total heap size”
+* Includes Callback Function
+
+UpDown Counter
+
+* Name: threadsActive
+* Unit: “1”
+* Description: “The total number of threads active”
+
+Asynchronous Gauge
+
+* Name: cpuUsage
+* Unit: “1”
+* Description: “Cpu usage percent”
+* Includes Callback Function
+
+**Common Attributes for Random based metrics**
+
+These are Key Value pairs to be added on metrics.
+```
+{
+“signal”:     (string)“metric”
+“language”:   (string)<name of language used>
+“metricType”: (string)“random”
+}
+```
+
+#### Logs
+
+**Logs are OPTIONAL due to their status of preview upstream. Logs must not be tested against nor will the logging structure described be final. It is to be used as an example to construct and emit example logs.**
+Logs will be displayed in JSON format for simplicity and readability.  Logs will be sent out on three different occasions, requests, errors, and events. An example of each can be seen underneath this section of the document. We will try to keep logs consistent in all sample applications according to the Logs Data Model (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md) found in the opentelemetry-specification repository.
+
+Logs will contain the following fields:
+
+* Timestamp
+* TraceID
+* SpanID
+* SeverityText (or LogLevel)
+* Body
+
+```
+Info Logs 
+
+// Info Logs are usually sent on application startup.
+
+{
+"level":"info",
+"msg":"Launching sampleapp: I am [sampleapp-svc] listening on port :8080 on all local IPs.",
+"time":"2022-06-28T11:31:38-07:00"
+}
+```
+
+```
+Event Log
+
+// Upon any event logs are sent out.
+
+{
+"event":"invoke","level":"info","msg":"sampleapp was invoked",
+"remote":"127.0.0.1:63875","time":"2022-06-27T15:58:35-07:00",
+"traceID":"1-62ba361b-0dc302c06b6bc21c2b2152a4"
+}
+```
+
+#### Traces
+
+At minimum, four GET requests endpoints must be present in each sample app. 
+The first endpoint will serve as a health check to confirm that the app has successfully ran.
+The second endpoint will make an HTTP GET request to aws.amazon.com (http://aws.amazon.com/). It will generate Traces and if applicable, Metrics.
+The third endpoint will make a request to AWS S3 and if credentials are provided, it will list all S3 buckets in the provided users S3.
+The fourth endpoint will make GET request to all other sample app endpoints that are configured. If no other sample apps are configured (leaf case), it will make a request to www.amazon.com (http://www.amazon.com/). 
+
+1. GET - /
+2. GET - /outgoing-http-call
+3. GET - /aws-sdk-call
+    
+4. GET - /outgoing-sampleapp
+
+An AWS X-Ray Trace ID should also be returned at the end of each request.
```

Comment:
All of these endpoints return an x-ray id as a response?

Reply:
Yes I believe so